wp_api_integration_tests/tests/test_app_notifier_immut.rs

```diff
@@ -0,0 +1,63 @@
+use async_trait::async_trait;
+use serial_test::parallel;
+use std::sync::{
+    Arc,
+    atomic::{AtomicBool, Ordering},
+};
+use wp_api::{
+    WpApiClient, WpAppNotification, WpAppNotifier, WpAuthentication, WpErrorCode,
+    middleware::WpApiMiddlewarePipeline, reqwest_request_executor::ReqwestRequestExecutor,
+    users::UserListParams,
+};
+use wp_api_integration_tests::{AssertWpError, test_site_url};
+
+#[tokio::test]
+#[parallel]
+async fn test_notification_unauthorized_request() {
+    let notifier: Arc<FooAppNotifier> =
+        FooAppNotifier::new(|notification| notification == WpAppNotification::UnauthorizedRequest)
+            .into();
+    api_client_as_unauthenticated_with_notifier(notifier.clone())
+        .users()
+        // Edit context requires authentication
+        .list_with_edit_context(&UserListParams::default())
+        .await
+        .assert_wp_error(WpErrorCode::ForbiddenContext);
+    assert!(
+        notifier.assertion.load(Ordering::Relaxed),
+        "Failed to notify with `WpAppNotification::UnauthorizedRequest`"
+    );
+}
+
+fn api_client_as_unauthenticated_with_notifier(notifier: Arc<FooAppNotifier>) -> WpApiClient {
+    WpApiClient::new(
+        test_site_url(),
+        WpAuthentication::None,
```

Comment:
This test set up is not suitable for the purpose of detecting an invalid application password. We should pass a hard-coded username and _invalid_ application password pair here.

Reply:
Addressed in d3b589560d191f146b1da64a8c91282f91b3225d.